common/src/main/java/org/keycloak/common/Version.java

```diff
@@ -32,7 +32,7 @@ public class Version {
     public static String NAME_HTML;
     public static String VERSION;
     public static String VERSION_KEYCLOAK;
-    public static String RESOURCES_VERSION;
+    public volatile static String RESOURCES_VERSION;
```

Comment:
This field is expected to be set once upon start (by a single thread) and then used, no concurrency is expected. Why to mark it volatile?

Reply:
I agree this would usually work, yet there are not many guarantees in Java when accessing a shared field without locking or synchronization. Making it volatile will hint to the developers that it is updated from other places as well.

Reply:
I believe this is unrelated to this PR, and the benefit is debatable. Could you please remove this from this PR to be discussed separately?

Reply:
OK, removed from this PR.